src/app/collective-rewards/metrics/hooks/useGetTimeUntilNextCycle.ts

```diff
@@ -0,0 +1,28 @@
+import { CycleTimeKeeperAbi } from '@/lib/abis/v2/CycleTimeKeeperAbi'
+import { BackersManagerAddress } from '@/lib/contracts'
+import { AVERAGE_BLOCKTIME } from '@/lib/constants'
+import { useQuery } from '@tanstack/react-query'
+import { readContract } from 'wagmi/actions'
+import { config } from '@/config'
+
+export const useGetTimeUntilNextCycle = (timestamp: bigint) => {
```

Comment:
have a look at this hook https://github.com/RootstockCollective/dao-frontend/pull/438/files#diff-ac810d5c5d8e70583cab271d6f69ec432b470e3ef9ee878023a2172d2bcff872. Maybe you could use that to minimise the footprint for each contract "service".